pkg/pipelinerunmetrics/metrics.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pipelinerunmetrics
+
+import (
+	"context"
+	"sync"
+
+	"github.com/tektoncd/chains/pkg/chains"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/metrics"
+)
+
+var (
+	sgCount = stats.Float64(chains.PipelineRunSignedName,
+		chains.PipelineRunSignedDesc,
+		stats.UnitDimensionless)
+
+	sgCountView *view.View
+
+	plCount = stats.Float64(chains.PipelineRunUploadedName,
+		chains.PipelineRunUploadedDesc,
+		stats.UnitDimensionless)
+
+	plCountView *view.View
+
+	stCount = stats.Float64(chains.PipelineRunStoredName,
+		chains.PipelineRunStoredDesc,
+		stats.UnitDimensionless)
+
+	stCountView *view.View
+
+	mrCount = stats.Float64(chains.PipelineRunMarkedName,
+		chains.PipelineRunMarkedDesc,
+		stats.UnitDimensionless)
+
+	mrCountView *view.View
+)
+
+// Recorder holds keys for Tekton metrics
+type Recorder struct {
+	initialized bool
+}
+
+// We cannot register the view multiple times, so NewRecorder lazily
+// initializes this singleton and returns the same recorder across any
+// subsequent invocations.
+var (
+	once           sync.Once
+	r              *Recorder
+	errRegistering error
+)
+
+// NewRecorder creates a new metrics recorder instance
+// to log the PipelineRun related metrics
+func NewRecorder(ctx context.Context) (*Recorder, error) {
+	logger := logging.FromContext(ctx)
+	once.Do(func() {
+		r = &Recorder{
+			initialized: true,
+		}
+		errRegistering = viewRegister()
+		if errRegistering != nil {
+			r.initialized = false
+			logger.Errorf("View Register Failed ", r.initialized)
+			return
+		}
+	})
+
+	return r, errRegistering
+}
+
+func viewRegister() error {
+	sgCountView = &view.View{
+		Description: sgCount.Description(),
+		Measure:     sgCount,
+		Aggregation: view.Count(),
+	}
+
+	plCountView = &view.View{
+		Description: plCount.Description(),
+		Measure:     plCount,
+		Aggregation: view.Count(),
+	}
+
+	stCountView = &view.View{
+		Description: stCount.Description(),
+		Measure:     stCount,
+		Aggregation: view.Count(),
+	}
+
+	mrCountView = &view.View{
+		Description: mrCount.Description(),
+		Measure:     mrCount,
+		Aggregation: view.Count(),
+	}
+	return view.Register(
+		sgCountView,
+		plCountView,
+		stCountView,
+		mrCountView,
+	)
+}
+
+func viewUnregister() {
```

Comment:
I don't see this being used.

Reply:
I was testing for any use case which may require unregistering, but didn't find any. So removed it.